### PR TITLE
[TensorBoard]  Integrate TensorBoard in submit-job-v2 plugin with external storage

### LINKED ISSUE #3147: [TensorBoard]  Clone jobs from submit-job-v2 plugin with TensorBoard config
- Clone jobs from submit-job-v2 plugin with new TensorBoard config.

### LINKED ISSUE #3157: [TensorBoard] New TensorBoard entry on job-detail page
- New TensorBoard entry on job-detail page
This entry will parse the extra field of protocol and generate web address.

### PR DESCRIPTION
**Pull request by TongboZhang**
Integrate TensorBoard in submit-job-v2 plugin with external storage.
TensorBoard config will be stored on the extra filed of protocol as the new design.

New Feature
- support NFS and HDFS;
- preview TensorBoard config changes in real time; 

Bug Fix
- Jobs can be correctly cloned from job-detail page ( Fixes #3147 ).
- New TensorBoard entry will parse the extra field of protocol and generate web address ( Fixes #3157 ).

### CODE REVIEW

**Comment by TongboZhang** [comment]
#3147

**Comment by TongboZhang** [comment]
#3156

**Comment by TongboZhang** [comment]
Fixes #3147

**Comment by TongboZhang** [comment]
image

**Comment by TongboZhang** [comment]
image

File: contrib/submit-job-v2/src/App/ProtocolForm.tsx [hunk 1]
```diff
@@ -571,46 +661,280 @@ export default class ProtocolForm extends React.Component<IProtocolProps, IProto
...
+    this.setState({
+      protocol,
+      protocolYAML: yaml.safeDump(protocol),
+      tensorBoardConfig,
+    });
+    await this.addTensorBoardTaskRole();
+    await this.injectCommand();
+  }
+
+  private addTensorBoardTaskRole = async () => {
```
Comment (abuccts) [inline contrib/submit-job-v2/src/App/ProtocolForm.tsx]: does `addTensorBoardTaskRole` need to be `async`? doesn't wait `Promise` in it.

File: contrib/submit-job-v2/src/App/ProtocolForm.tsx [hunk 2]
```diff
@@ -571,46 +661,280 @@ export default class ProtocolForm extends React.Component<IProtocolProps, IProto
...
     protocol.taskRoles[tensorBoardName].resourcePerInstance.ports[tensorBoardPort] = 1;
-    protocol.extras = { tensorBoardStr: randomStr };
+
+    this.setState({
+      protocol,
+      protocolYAML: yaml.safeDump(protocol),
+    });
+  }
+
+  private deleteTensorBoardConfig = async () => {
```
Comment (abuccts) [inline contrib/submit-job-v2/src/App/ProtocolForm.tsx]: same, does `deleteTensorBoardConfig` need to be `async`?

File: contrib/submit-job-v2/src/App/ProtocolForm.tsx (hunk 1 above)
Comment (TongboZhang) [reply contrib/submit-job-v2/src/App/ProtocolForm.tsx]: OK, I will rebuild the code.

File: contrib/submit-job-v2/src/App/ProtocolForm.tsx [hunk 3]
```diff
@@ -323,13 +334,10 @@ export default class ProtocolForm extends React.Component<IProtocolProps, IProto
               />
             </Stack>
             <Stack className={styles.item}>
-              <Toggle
-                label="Enable TensorBoard"
-                checked={this.state.enableTensorBoard}
-                onChange={this.toggleTensorBoard}
-                inlineLabel={true}
+              <TensorBoard
```
Comment (abuccts) [inline contrib/submit-job-v2/src/App/ProtocolForm.tsx]: change the position with parameters stack?

File: contrib/submit-job-v2/src/App/ProtocolForm.tsx [hunk 4]
```diff
@@ -158,8 +160,6 @@ interface IProtocolState {
   loading: boolean;
   showParameters: boolean;
   showEditor: boolean;
-  logPath: string;
-  enableTensorBoard: boolean;
```
[the remark this answers is not in the document]
Comment (TongboZhang) [reply contrib/submit-job-v2/src/App/ProtocolForm.tsx]: I have removed tensorBoardConfig here since it is useless in this component.

**Pull request merged**